packages/analysis-report/src/AnalysisResult.js

```diff
@@ -79,14 +82,19 @@ const createMarkButton = ( {
  * @returns {ReactElement} The rendered AnalysisResult component.
  */
 const AnalysisResult = ( { markButtonFactory, ...props } ) => {
+	const [ isOpen, setIsOpen ] = useState( false );
+
+	const closeModal = useCallback( () => setIsOpen( false ), [] );
+	const openModal = useCallback( () => setIsOpen( true ), [] );
+
 	markButtonFactory = markButtonFactory || createMarkButton;
 	const { id, marker, hasMarksButton } = props;
 
 	let marksButton = null;
 	if ( ! areMarkButtonsHidden( props ) ) {
 		marksButton = markButtonFactory(
 			{
-				onClick: props.onButtonClickMarks,
+				onClick: props.shouldUpsellHighlighting ? openModal : props.onButtonClickMarks,
```

Comment:
When in Elementor and only Yoast SEO free is active, we set `isOpen` state to true. If `isOpen` is true, we render the highlighting upsell modal